src/modules-load/modules-load.c

```
@@ -71,7 +90,27 @@ static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent
                 if (strchr(COMMENTS, *line))
                         continue;
 
-                k = module_load_and_warn(ctx, line, true);
+                RET_GATHER(r, send_module_to_load(sock, line));
+        }
+
+        return r;
+}
+
+static int run_prober(int sock) {
+        _cleanup_(sym_kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        char buffer[NAME_MAX + 1];
+        ssize_t bytes_received;
+
+        int r = module_setup_context(&ctx);
+        if (r < 0)
+                log_error_errno(r, "Failed to initialize libkmod context: %m");
+
+        while ((bytes_received = recv(sock, buffer, sizeof(buffer), 0)) > 0) {
```

please handle errors from recv() in some reasonable way. i.e. some errors are unexpected and we should log about them, other errors are expected (i.e. connection termination) and should not be ogged about

---

Reworked - but I'm still reasoning on how to handle EINTR. I'll leave this point open for now.

---

The `EINTR` error is ignored, while any other causes the receive loop to fail.